### PR TITLE
[move] Additional Prover specs for Coin module (+ Move version bump)

### PR DESCRIPTION
**Pull request by awelc**
This PR adds (working) Prover specs to the Coin module to serve as a template for others attempting to write more specs. The Move version bump is necessary to bring a relevant fix (https://github.com/move-language/move/pull/736) from the core Move side.

### CODE REVIEW

File: crates/sui-framework/sources/coin.move [hunk 1]
```diff
@@ -249,6 +310,14 @@ module sui::coin {
         balance::increase_supply(&mut cap.total_supply, value)
     }
 
+    spec mint_balance {
```
Comment (emmazzz) [inline crates/sui-framework/sources/coin.move:313]: The specs here and for `mint` seem very similar. Maybe use `schema`? https://github.com/move-language/move/blob/main/language/move-prover/doc/user/spec-lang.md#schemas
Same for `burn` and `burn_`.

File: crates/sui-framework/sources/coin.move [hunk 2]
```diff
@@ -271,13 +348,11 @@ module sui::coin {
         burn(c, coin);
     }
 
-    // === Update coin metadata ===
-
-    /// Update name of the coin in `CoinMetadata`
-    public entry fun update_name<T>(
```
Comment (emmazzz) [inline crates/sui-framework/sources/coin.move:277]: Why is this function deleted?

File: crates/sui-framework/sources/coin.move (hunk 1 above)
Comment (emmazzz) [reply crates/sui-framework/sources/coin.move:313]: Does it help if you add `<T>` to the schema signature too? Like
```
spec schema MintSpecs<T> {
    cap: TreasuryCap<T>;
    ...
}
```
There are some examples in Diem Framework specs that might be helpful: https://github.com/move-language/move/blob/main/language/documentation/examples/diem-framework/move-packages/DPN/sources/Diem.move#L453

Comment (awelc) [reply crates/sui-framework/sources/coin.move:313]: Yeah, I found them :-) Hence my comment deletion :-) Getting there!

Comment (awelc) [reply crates/sui-framework/sources/coin.move:320]: Done!

**Review by emmazzz — APPROVED** [review]

**Pull request merged**